src/lib/CLValue/Abstract.ts

```diff
@@ -34,14 +34,17 @@ export const resultHelper = <T, E>(
 };
 
 export abstract class CLType {
-  abstract toString(): string;
-  abstract toJSON(): any;
-  abstract linksTo: any;
+  abstract linksTo: string;
   abstract tag: CLTypeTag;
-
   toBytes(): Uint8Array {
     return Uint8Array.from([this.tag]);
   }
+  toString(): string {
+    return this.linksTo;
+  }
+  toJSON(): unknown {
```

Comment:
Just curious how could this return an `unknown` type if this returns toString in any case.

Reply:
its beeing overwritten in some places like for example in [`CLMap`](https://github.com/casper-ecosystem/casper-js-sdk/blob/dev/src/lib/CLValue/Map.ts#L63-L70)

Reply:
This seems to me unclear how CLMap::toJSON() can return any type ? isn't it a string too ?

I would remove that `unknow` return type here anyway and let it infer the type, it's a bit confusing tbh. `unknow` is mainly forbidden to use or when you don't have the choice when casting as, but I would not use it in a function definition.

Reply:
no, `CLMap::toJSON()` will return object, I send you an example in a link above.
In most cases it returns string, but this is not a rule. Sometimes its overwritten to return proper `cl_type`.

For me its more adequate to use `unknown` here instead of any, look at this documentation https://devblogs.microsoft.com/typescript/announcing-typescript-3-0/#the-unknown-type

Reply:
IMO, the return type of `toJSON` method should be `any` or `unknow` because there is `CLType::Any`.
But I think the `toJSON` method returns this object in general.

```json
{
  "bytes": <hex string representation of bytes>,
  "parsed": <parsed data:unknown>,
  "cl_type": <string represent of CLType>
}
```
So I think the return type of the toJSON should be

```typescript
type ToJSON = {
 bytes: string;
 data: unknown;
 cl_type: CLType Enum string
}
```

Reply:
To many `toJSON` methods, just confused.